frontend/src/pages/surveys/components/step/render-steps.tsx

```diff
@@ -0,0 +1,172 @@
+import { useAppDispatch, useCallback } from '#libs/hooks/hooks.js';
+import { type StepsType } from '#packages/survey/libs/types/types.js';
+import { SurveySteps } from '#packages/survey/survey.js';
+import {
+  FEELINGS_CATEGORIES,
+  GOALS_CATEGORIES,
+  JOURNALING_EXPERIENCE_CATEGORIES,
+  MEDITATION_EXPERIENCE_CATEGORIES,
+  PREFERENCES_CATEGORIES,
+  WORRIES_CATEGORIES,
+} from '#pages/surveys/libs/constants/constants.js';
+import { actions } from '#slices/survey/survey.js';
+
+import { Step } from './step.js';
+
+type StepsProperties = {
+  currentStep: StepsType;
+  onNextStep: () => void;
+  onPreviousStep: () => void;
+  onSubmit: () => void;
+};
+
+const RenderSteps: React.FC<StepsProperties> = ({
+  currentStep,
+  onNextStep,
+  onPreviousStep,
+  onSubmit,
+}): React.ReactNode => {
+  const dispatch = useAppDispatch();
+
+  const setPreference = useCallback(
+    (preferences: string[] | string): void => {
+      if (Array.isArray(preferences)) {
+        dispatch(actions.setPreferences(preferences));
+      }
+    },
+    [dispatch],
+  );
+
+  const setFeelings = useCallback(
+    (feelings: string[] | string): void => {
+      if (Array.isArray(feelings)) {
+        dispatch(actions.setFeelings(feelings));
+      }
+    },
+    [dispatch],
+  );
+
+  const setGoals = useCallback(
+    (goals: string[] | string): void => {
+      if (Array.isArray(goals)) {
+        dispatch(actions.setGoals(goals));
+      }
+    },
+    [dispatch],
+  );
+
+  const setWorries = useCallback(
+    (worries: string[] | string): void => {
+      if (Array.isArray(worries)) {
+        dispatch(actions.setWorries(worries));
+      }
+    },
+    [dispatch],
+  );
+
+  const setMeditationExperience = useCallback(
+    (meditationExperience: string[] | string): void => {
+      if (typeof meditationExperience === 'string') {
+        dispatch(actions.setMeditationExperience(meditationExperience));
+      }
+    },
+    [dispatch],
+  );
+
+  const setJournalingExperience = useCallback(
+    (journalingExperience: string[] | string): void => {
+      if (typeof journalingExperience === 'string') {
+        dispatch(actions.setJournalingExperience(journalingExperience));
+      }
+    },
+    [dispatch],
+  );
+
+  switch (currentStep) {
```

Comment:
That switch case looks so repetitive. Can you create mapper? At least you can create callback that will take unique params like `stepCategories` and `question`